tests/ga/test_extension.py

```diff
@@ -1481,72 +1481,50 @@ def test__extension_upgrade_failure_when_prev_version_disable_fails(self, patch_
         test_data, exthandlers_handler, protocol = self._set_up_update_test_and_update_gs(patch_get_disable_command,
                                                                                           *args)
 
-        exthandlers_handler.run()  # Download the new update the first time, and then we patch the download method.
-
-        with patch('azurelinuxagent.common.protocol.restapi.Protocol.download_ext_handler_pkg') as patch_download:
-            loop_run = 5
-            for x in range(loop_run):
-                exthandlers_handler.run()
-
-            self.assertEqual(loop_run + 1, patch_get_disable_command.call_count)  # counting the earlier run done to
-            # download the new update
-            self.assertEqual(0, patch_download.call_count)  # The download should never be called.
-
-        # On the next iteration, update should not be retried
-        self._assert_handler_status(protocol.report_vm_status, "NotReady", expected_ext_count=0, version="1.0.1")
-
-    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_disable_command')
-    def test__extension_upgrade_failure_when_prev_version_disable_fails_and_recovers(self, patch_get_disable_command,
```

Comment:
I dont know how this test was recovering before (maybe by using the is_upgrade flag) - `test__extension_upgrade_failure_when_prev_version_disable_fails_and_recovers`, but maybe you could keep it just to ensure that we have this case tested too.

Everything else LGTM!

Reply:
Good point, this test was "recovering" by simply running the exthandler sequence for the same goal state, which is not valid anymore. However, there is value in having a test that exercises the recover scenario of the upgrade sequence (this time on a new incarnation). I'll add that test. Thanks!